### PR TITLE
Arrays and accessing array properties

### PR DESCRIPTION
**Pull request by ggetz**
Added Array expressions, and tests to evaluate arrays themselves and accessing array properties in features.

Operations on arrays follow JS behavior, for example, `[1, 2, 3] * 4 === NaN` and `([1,2,3] === [1,2,3]) === false`

### CODE REVIEW

File: Source/Scene/Expression.js [hunk 1]
```diff
@@ -273,6 +273,12 @@ define([
             node = new Node(ExpressionNodeType.CONDITIONAL, '?', left, right, test);
         } else if (ast.type === 'MemberExpression') {
             node = parseMemberExpression(expression, ast);
+        } else if (ast.type === 'ArrayExpression') {
+            var val = [];
+            for (var i=0; i < ast.elements.length; i++) {
```
Comment (pjcozzi) [inline Source/Scene/Expression.js]: Minor whitespace.  Throughout this file and in general, use something like: `for (var i = 0;`

**Comment by pjcozzi** [comment]
This looks good, just that trivial comment.

**Comment by ggetz** [comment]
OK, thanks! Updated.

**Comment by pjcozzi** [comment]
There are several instances in that file, which is why the comment said "Throughout this file and in general"

**Comment by ggetz** [comment]
OK, sorry, I fixed all the occurrences I found.

**Pull request merged**